bin/import_xml2obd.py

```diff
@@ -157,21 +185,105 @@ def get_preset(name):
+        except ckan.logic.NotFound:
+            try:
+                site.action.package_create(**rec)
+            except ckan.logic.ValidationError:
+                traceback.print_exc()
+                print(rec)
+                continue
+        except ckan.logic.ValidationError:
+            traceback.print_exc()
+            print(rec)
+            continue
+        res = rec['resources'][0]
+        source = resource_directory + '/' + res['url'].split('/')[-1]
+        with open(source) as f:
+            try:
+                rc = site.action.resource_patch(
+                    id=res['id'],
+                    upload=(os.path.basename(source), f))
+            except ckanapi.errors.CKANAPIError:
+                traceback.print_exc()
+                print(rec)
+                print("failed upload " + os.path.basename(source))
+                continue
+
+        print("updated for " + os.path.basename(source))
+
+
 def main():
     global presets, audience, canada_resource_type,canada_subject
     global canada_resource_language
+    canada_resource_format = None
+
+    # upload the generated json to remote site and upload the file
+    if sys.argv[1] =='upload':
+        return upload_resources(*sys.argv[2:])
+
     refs, extras = read_xml(sys.argv[1])
     read_organizations_from_json(sys.argv[2])
-    presets = read_presets(sys.argv[3])
+    presets = read_presets(sys.argv[3])  # TODO: read from doc.yaml
     audience = get_preset('canada_audience')
     canada_resource_type = get_preset('canada_resource_type')
     canada_subject = get_preset('canada_subject')
     canada_resource_language = get_preset('canada_resource_language')
+    canada_resource_format = map(lambda x:x['value'],
+                                 get_preset('canada_resource_format')['choices'])
 
     ds = xml_obd_mapping(refs, xml2obd)
     ds['collection'] = 'publication'
+    release_date = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    if not ds.get('date_published'):
+        ds['date_published'] = release_date
+    ds['id'] = str(uuid.uuid5(uuid.NAMESPACE_URL,
+                   'http://dev.obd.ca/' + sys.argv[1]))
+    ds['name'] = ds['id']
+    ds['title'] = ds['title_translated']['en']
+    ds['state'] = 'active'
+    ds['type'] = 'doc'
+    ds['license_id'] = "ca-ogl-lgo"
+    if not ds.get('owner_org', None):
+        ds['owner_org'] = "A0F0FCFC-BC3B-4696-8B6D-E7E411D55BAC"
+    if not ds.get('keywords', None):
+        ds['keywords']={'en':['statcan'], 'fr':['statcan']}
+    if not ds.get('subject', None):
+        ds['subject'] = ["information_and_communications"]
+    if not ds.get('notes_translated', None):
+        ds['notes_translated'] = {'en':'n/a', 'fr':'n/a'}
+    if not ds['notes_translated'].get('fr', None):
+        ds['notes_translated']['fr'] = ds['notes_translated']['en']
+    if not ds['title_translated'].get('fr', None):
+        ds['title_translated']['fr'] = ds['title_translated']['en']
```

Comment:
I'm going to make the notes optional for this type instead. I don't like filling in "n/a" unless we know that something is actually not applicable